### PR TITLE
New Plugin Added

### PR DESCRIPTION
**Pull request by bdt0-rocklee**
CRTG(CS Research Topic Generator) plugin added. Updated the Commands list, Version file.

### CODE REVIEW

**Comment by woodruffw** [comment]
Please add `cs_research_topicgen` to the `Gemfile` so that Bundler picks it up.

**Comment by woodruffw** [comment]
Looks good, I'll test and merge in a bit.

File: plugins/crtg.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	use_blacklist
+
+	def usage
+		'!crtg - Generate CS Research Topics.'
+	end
+
+	def match?(cmd)
+		cmd =~ /^(!)?crtg$/
+	end
+
```
Comment (woodruffw) [inline plugins/crtg.rb]: There's a line missing here. You need something like this:

```ruby
match /crtg$/, method: :crtg
```

File: plugins/crtg.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	def crtg()
```
Comment (woodruffw) [inline plugins/crtg.rb]: Missing argument `m`.

**Comment by bdt0-rocklee** [comment]
:+1:

**Comment by woodruffw** [comment]
Mind squashing this down into a single commit?

**Comment by woodruffw** [comment]
This needs to be squashed.

File: plugins/cryptogen.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#  -*- coding: utf-8 -*-
+#  cryptogen.rb
+#  Author: slackR
+#  ------------------------
+#  A Cinch plugin that generates worthy Crypto Research Topics for yossarian-bot.
+#  ------------------------
+#  This code is licensed by slackR under the MIT License.
+#  http://opensource.org/licenses/MIT
+
+require	'research_topicgen'
```
Comment (woodruffw) [inline plugins/cryptogen.rb]: Tabs for indentation, not for spacing.

**Comment by woodruffw** [comment]
Spacing needs to be fixed and merge conflict needs to be resolved, but after that this should be good :+1:

**Comment by bdt0-rocklee** [comment]
Will take care of that soon. :grin:

File: COMMANDS.md [hunk 4]
```diff
@@ -29,6 +29,7 @@ Command Syntax | Plugin class | Description | Working?
 `!cbsg` | CBSG | Spew some corporate bullshit from the [Corporate Bullshit Generator](http://cbsg.sourceforge.net/cgi-bin/live). | Yes
 `!channel <commands>` | ChannelAdmin | Administrate the current channel. [See below](#chanadmin). | Yes
 `!clickbait` | Clickbait | Generate a clickbait-y title. | Yes
+`!cstopic <cs|system|crypto|random>` | Topic | Generate worthy CS,System,Crypto Research Topics. | Yes
```
Comment (woodruffw) [inline COMMANDS.md]: This should probably just stay as `!topic`, because `!cstopic cs` sounds strange.

File: plugins/cstopic.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+#  A Cinch plugin that generates CS research topics for yossarian-bot.
+#  ------------------------
+#  This code is licensed by slackR under the MIT License.
+#  http://opensource.org/licenses/MIT
+
+require 'research_topicgen'
+
+require_relative 'yossarian_plugin'
+
+class CSTopic < YossarianPlugin
```
Comment (woodruffw) [inline plugins/cstopic.rb]: Pluralize: `CSTopics`

File: plugins/cstopic.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	include Cinch::Plugin
+	use_blacklist
+
+	def usage
+	  '!cstopic <cs|system|crypto> - Generate Worthy Research Topics.'
```
Comment (woodruffw) [inline plugins/cstopic.rb]: `!topic`

File: plugins/cstopic.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	end
+
+	match /topic (.+)/, method: :cstopic
+
+	def match?(cmd)
+		cmd =~ /^(!)?cstopic$/
```
Comment (woodruffw) [inline plugins/cstopic.rb]: `!topic`

File: plugins.yml [hunk 8]
```diff
@@ -16,6 +16,7 @@ plugins:
     - Clickbait
     - CommandHelp
     - CodeEval
+    - CSTopic
```
Comment (woodruffw) [inline plugins.yml]: `CSTopics`

File: config.yml.example [hunk 9]
```diff
@@ -28,4 +28,4 @@ servers:
         channels:
             - '#bots'
         disabled_plugins: []
-
+
```
Comment (woodruffw) [inline config.yml.example]: Trailing tab here needs to be deleted.

File: plugins/cstopic.rb [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+		  m.reply ResearchTopicGen.system, true
+		elsif search =~	/crypto/
+		  m.reply ResearchTopicGen.crypto, true
+		elsif search =~ /random/
+		  m.reply ResearchTopicGen.send([:cs, :system, :crypto].sample)
+		else
+		  m.reply "Research Topics for #{topic} can't be generated.", true
+		end
+	end
+end
```
Comment (woodruffw) [inline plugins/cstopic.rb]: Needs a trailing newline.

File: plugins/cstopic.rb [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+	end
+
+	match /topic (.+)/, method: :cstopic
+
+	def match?(cmd)
+		cmd =~ /^(!)?cstopic$/
+	end
+
+	def cstopic(m, search)
+
```
Comment (woodruffw) [inline plugins/cstopic.rb]: Remove extra l

File: config.yml.example [hunk 12]
```diff
@@ -28,4 +28,3 @@ servers:
         channels:
             - '#bots'
         disabled_plugins: []
-
```
Comment (woodruffw) [inline config.yml.example]: Trailing newline needed.

**Comment by woodruffw** [comment]
Once the comments above are applied, I can merge this.

File: plugins/cstopic.rb [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+require 'research_topicgen'
+
+require_relative 'yossarian_plugin'
+
+class CSTopics < YossarianPlugin
+	include Cinch::Plugin
+	use_blacklist
+
+	def usage
+	  '!topic <cs|system|crypto> - Generate Worthy Research Topics.'
```
Comment (woodruffw) [inline plugins/cstopic.rb]: "Generate worthy research topics."

File: COMMANDS.md [hunk 14]
```diff
@@ -71,6 +71,7 @@ Command Syntax | Plugin class | Description | Working?
 `!taco` | TacoRecipes | Get a random taco recipe, courtesy of the [Taco Randomizer](http://taco-randomizer.herokuapp.com/). | Yes
 `!theo` | TheoQuotes | Get a random Theo De Raadt quote. | Yes
 `!tr <text>` | GoogleTranslate | Translate text to English with [Google Translate](https://translate.google.com). | Deprecated (API removed)
+`topic <cs|system|crypto|random>` | CSTopics | Generate worthy CS,System,Crypto Research Topics. | Yes
```
Comment (woodruffw) [inline COMMANDS.md]: Put spaces between commas.

**Comment by woodruffw** [comment]
:+1:

**Pull request merged**